packages/features/ee/sso/lib/saml.ts

```diff
@@ -57,3 +58,49 @@ export const samlTenantProduct = async (prisma: PrismaClient, email: string) =>
     product: samlProductID,
   };
 };
+
+export const canAccess = async (
```

Comment:
This method has been altered to send back appropriate messages in each case. This message will be displayed on the UI when TRPC throws an error https://github.com/calcom/cal.com/pull/4938/files#diff-015067416d87bbb0a3405527c31019007de06a40b377883a98b6e32623c2d0e5R25